core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -515,7 +515,7 @@ protected Scope visitRefreshMaterializedView(RefreshMaterializedView refreshMate
                 throw semanticException(TABLE_NOT_FOUND, refreshMaterializedView, "Storage Table '%s' for materialized view '%s' does not exist", storageName, name);
             }
 
-            QualifiedObjectName targetTable = createQualifiedObjectName(session, refreshMaterializedView, storageName.get());
+            QualifiedObjectName targetTable = metadata.redirectTable(session, createQualifiedObjectName(session, refreshMaterializedView, storageName.get()));
```

Comment:
In case of error, will error message refer to redirected table, e.g:
```
SELECT * FROM table_to_be_redirected
```
would cause e.g
```
Cannot analyze redirected_table
```
?

Reply:
yeah, changes along the lines of #7134 would be helpful to propagate this information properly, when it's checked in.

for now, I think redirected table in error message helps keep the error message behavior consistent. i.e. any future errors (say during execution) will only refer to the target as the source won't be available. I agree though that it can be a bit confusing. wdyt?

Reply:
https://github.com/trinodb/trino/pull/7134 is only partially helpful as warning/events might not be shown by all tools.

> or now, I think redirected table in error message helps keep the error message behavior consistent. i.e. any future errors (say during execution) will only refer to the target as the source won't be available.

I'm not sure we actually refer to table names during execution. We do during optimization, but even there we already use "virtual" table names for pushed down `TableHandles`.
I think the biggest confusion will actually come during analysis as error messages won't match queries.

I also look at a way to reduce amount of changes required by new SPI. Currently, it's pretty extensive and touches a lot of components. Potentially, transparent redirection within `getTableHandle` could reduce scope of changes.